.github/actions/download-builds/action.yml

```diff
@@ -0,0 +1,11 @@
+name: Download and Extract Build Artifacts
+description: Download build artifacts for all the packages and extract them to the correct location
+
+runs:
+  using: "composite"
```

Comment:
Why this action is composite and the second one isn't?

Reply:
the build and upload can be a job of its own (reusable workflow)
reusable action is a `step` that encapsulates multiple `steps`
reusable workflow is a `job` that encapsulates multiple `jobs`

the reason I chose going with a reusable workflow for the `build-and-upload` is that in case we need the build, I'm saving the consumer from doing `checkout` and `setup`. the consumer can just do:
```
  build:
    name: Build
    uses: ./.github/workflows/build-and-upload.yml
    secrets:
      npm_token: ${{ secrets.npm_token }}
```

if the `build-and-upload` was a reusable action (`composite`), it would have looked like this:
```
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
        with:
          fetch-depth: 0
      - name: Run Setup
        uses: ./.github/actions/setup
        with:
          npm_token: ${{secrets.npm_token}}
      - name: Run Build
        shell: bash
        run: yarn lerna run build --since=origin/master --include-dependencies
```

in probably 100% of the cases where consumers would need the build, they would just want it to happen, they don't need to interfere what happens with it